### PR TITLE
:books: Add design documentation

### PR DESCRIPTION
**Pull request by elbeno**

### CODE REVIEW

File: docs/paths.adoc [hunk 1]
```diff
@@ -0,0 +1,148 @@
+
+== Paths
+
+A `path` is a way to access a field or register. It is a compile-time list of
```
Comment (lukevalenty) [inline docs/paths.adoc:4]: suggestion: may be useful to point out the analogy with filesystem and url paths

File: docs/paths.adoc [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+Paths can be constructed using UDLs which are in the `groov::literals` namespace.
+
+[source,cpp]
+----
+using namespace groov::literals;
+
+auto p1 = "reg"_r;
+auto p2 = "field"_f;
+auto p3 = "group"_g;
```
Comment (lukevalenty) [inline docs/paths.adoc:17]: question: what is `_g` used for? i don't see how it is used with the current api.

File: docs/paths.adoc [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+----
+
+NOTE: The `_r` `_f` and `_g` literals are all equivalent; they are intended
```
Comment (lukevalenty) [inline docs/paths.adoc:20]: question: if i refer to a register using `_f`, will that result in a compilation error?

File: docs/read_write.adoc [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+w_spec["reg"_r] = 17;          // update value
+assert(w_spec["reg"_r] == 17); // retrieve value
+----
+
+A `write_spec` may be indexed with any path that is resolvable by a bound path.
+
+[source,cpp]
+----
+auto w_spec = grp / ("reg"_r = 42); // "reg" is bound
+w_spec["reg.field"_r] = 1;          // OK, assuming "reg" contains "field"
```
Comment (lukevalenty) [inline docs/read_write.adoc:43]: suggestion: use `_f` here since we are referring to a field.

**Review by lukevalenty — APPROVED** [review]

File: docs/paths.adoc (hunk 3 above)
Comment (elbeno) [reply docs/paths.adoc:20]: No - they're all the same.

File: docs/paths.adoc (hunk 2 above)
Comment (elbeno) [reply docs/paths.adoc]: I don't think it is used. I've removed it from the docs.

**Pull request merged**